### PR TITLE
Add trantor/1.0.0

### PR DESCRIPTION
**Pull request by an-tao**
Specify library name and version:  trantor/1.0.0

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/trantor/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+    
+    def _source_subfolder(self):
```
Comment (intelligide) [inline recipes/trantor/all/conanfile.py:30]:
```suggestion

    @property
    def _source_subfolder(self):
```

File: recipes/trantor/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "trantor-{}".format(self.version)
+        os.rename(extracted_dir, self._source_subfolder())
+    def _configure_cmake(self):
```
Comment (intelligide) [inline recipes/trantor/all/conanfile.py:37]:
```suggestion

    def _configure_cmake(self):
```

File: recipes/trantor/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+    def build(self):
```
Comment (intelligide) [inline recipes/trantor/all/conanfile.py:43]:
```suggestion

   def build(self):
```

File: recipes/trantor/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        cmake = self._configure_cmake()
+        cmake.build()
+    def package(self):
```
Comment (intelligide) [inline recipes/trantor/all/conanfile.py:46]:
```suggestion

    def package(self):
```

File: recipes/trantor/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        self.copy("License", src=self._source_subfolder(), dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+    def package_info(self):
```
Comment (intelligide) [inline recipes/trantor/all/conanfile.py:51]:
```suggestion

    def package_info(self):
```

File: recipes/trantor/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure()
+        return self._cmake
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+    def package(self):
+        self.copy("License", src=self._source_subfolder(), dst="licenses")
```
Comment (intelligide) [inline recipes/trantor/all/conanfile.py:47]:
```suggestion
         self.copy("License", src=self._source_subfolder, dst="licenses")
```

File: recipes/trantor/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        if self.options.shared:
+            del self.options.fPIC
+    
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "trantor-{}".format(self.version)
+        os.rename(extracted_dir, self._source_subfolder())
```
Comment (intelligide) [inline recipes/trantor/all/conanfile.py:36]:
```suggestion
        os.rename(extracted_dir, self._source_subfolder)
```

File: recipes/trantor/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
+from conans import ConanFile, CMake, tools
+import os
+
+class TrantorConan(ConanFile):
+    name = "trantor"
+    license = "https://raw.githubusercontent.com/an-tao/trantor/master/License"
```
Comment (intelligide) [inline recipes/trantor/all/conanfile.py:6]:
```suggestion
    license = "BSD-3-Clause"
```

**Review by intelligide — APPROVED** [review]

File: recipes/trantor/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    exports_sources = ["CMakeLists.txt"]
+    homepage = "https://github.com/an-tao/trantor"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Non-blocking I/O TCP network library based on +14/17"
+    topics = ("Network library", "C++14/17", "Non-blocking I/O")
```
Comment (uilianries) [inline recipes/trantor/all/conanfile.py:11]:
```suggestion
    topics = ("tcp-server", "non-blocking-io", "asynchronous", "networking")
```

File: recipes/trantor/all/test_package/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 3.1)
```
Comment (uilianries) [inline recipes/trantor/all/test_package/CMakeLists.txt:1]:
```suggestion
cmake_minimum_required(VERSION 3.4)
```

CXX_STANDARD 14 requires 3.4

File: recipes/trantor/all/test_package/test_package.cpp [hunk 11]
```diff
@@ -0,0 +1,8 @@
+#include <trantor/net/EventLoop.h>
+#include <iostream>
```
Comment (uilianries) [inline recipes/trantor/all/test_package/test_package.cpp:2]:
```suggestion
```

not used

**Review by uilianries — CHANGES_REQUESTED** [review]

File: recipes/trantor/all/conanfile.py [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+        self.requires("openssl/1.1.1h")
+        self.requires("c-ares/1.16.1")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
```
Comment (uilianries) [inline recipes/trantor/all/conanfile.py:28]:
```suggestion
            del self.options.fPIC
        if self.settings.compiler == "Visual Studio" and "MT" in self.settings.compiler.runtime and self.options.shared:
            raise ConanInvalidConfiguration("Trantor can not be built by MSVC and MT runtime as shared library.")
```

**Comment by uilianries** [comment]
@an-tao  The error is caused due Crypt32 missing:

>    libcrypto.lib(e_capi.obj) : error LNK2019: unresolved external symbol __imp_CertOpenStore referenced in function capi_open_store [C:\J\w\cci_PR-3117@2.conan\data\trantor\1.0.0__\build\19f67793fecef0b1b6dcb3d7f2815c46a3a040ba\source_subfolder\trantor.vcxproj]

The method CertOpenStore is part of wincrypt.h

**Pull request closed without merging**